rustls/src/crypto/tls12.rs

```diff
@@ -92,17 +92,28 @@ pub trait PrfSecret: Send + Sync {
 }
 
 pub(crate) fn prf(out: &mut [u8], hmac_key: &dyn hmac::Key, label: &[u8], seed: &[u8]) {
+    // Whether `out` is public or secret is up to the caller: the caller must zeroize
+    // `out` if it is secret, irrespective of whether the intermediate values are
+    // zeroized in this function.  Therefore there is no reason to zeroize them at this
+    // level.
```

Comment:
Sorry, I'm not sure I follow here. Can you help me out?

If the caller is extracting into `out` for the purpose of a secret usage, they can/should zeroize `out` once done, but this function isn't zeroizing the intermediate values and the caller can't either. I think what I'm missing is an assertion that the intermediates aren't helpful for cryptanalysis.